executor/README.MD

```diff
@@ -0,0 +1,104 @@
+Cook Executor
+=============
+
+The Cook executor is a custom executor written in Python.
+It replaces the default command executor in order to enable a number of features for both operators and end users.
+
+For Users
+---------
+
+The cook executor supports a number of (completely optional) features beyond the standard Mesos command executor.
+The executor sends the following information to the scheduler:
+1. progress update messages of the task (at intervals of `progress-poll-interval-ms` ms, see below)
+2. the exit code of the task
+3. the sandbox location of the task
+
+For Operators
+---------
+
+The Cook executor replaces the default command executor.
+It can either be installed on each agent, or downloaded on each agent via a `uri`.
+As the executor communicates with Mesos agent via HTTP, the agent must support the HTTP API and have it enabled.
+
+### Installation
+
+First, make sure `python3` is installed on each agent.
+
+Then, to install the executor executable (`cook-executor`) and all dependencies, run:
+
+```bash
+$ python setup.py install
+```
+
+Alternatively, [`pyinstaller`](https://github.com/pyinstaller/pyinstaller) can be used to build a single executable that is distributed via a URI.
+For example, to build the executable, run the following command from the `executor` folder:
+
+```bash
+# will create dist/cook-executor
+$ pyinstaller -F -n cook-executor -p cook cook/__main__.py
+```
+
+For development, a docker-compatible version of the cook executor can be built using:
+
+```bash
+# will create dist/cook-executor
+$ bin/build-cook-executor.sh
+```
+
+### Configuration
+
+Configuration for the executor should go under the `:executor` key of the edn configuration file for the cook scheduler.
+
+For example:
+
+```clojure
+{...
+ :executor {:command "cook-executor"
+            :log-level "INFO"
+            :max-message-length 512
+            :progress-poll-interval-ms 1000
+            :uri {:cache true
+                  :executable true
+                  :extract false
+                  :value "file:///path/to/cook-executor"}}
+ ...}
+```
+
+Supported configuration options:
+
+| option | type | description |
+|--------|------|-------------|
+| `:command` | string | A string containing the command executed on the mesos agent to launch the cook executor. If the executor is installed using the instructions above, the default value of `cook-executor` should not need to be changed. |
+| `:log-level` | string | The log level for the executor process. Defaults to "INFO" |
+| `:max-message-length` | long | The maximum length for the custom failure message set by a task via the executor HTTP API. The default is 512. |
```

Comment:
I meant Mesos's executor HTTP api which we are using to interact with the scheduler.